src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -757,6 +865,20 @@ protected static void ValidateBufferArguments(byte[] buffer, int offset, int cou
             }
         }
 
+        private static void ValidateReadAtLeastArguments(int bufferLength, int minimumBytes)
+        {
+            if (minimumBytes < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.minimumBytes, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (bufferLength < minimumBytes)
+            {
+                // TODO: pick the right exception message here
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.minimumBytes);
```

Comment:
Is `ArgumentOutOfRangeException` the most appropriate for this check?

Reply:
I _think_ so. This is the exception that is thrown directly above this method:

https://github.com/dotnet/runtime/blob/20cd393c1e577f0ff53cdc3a00edc3bcfc68ab7f/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L754-L757

in this case, `minimumBytes` is "out of the valid range" of `0 - buffer.Length`.